chore(user-guide): add add application context classloader switch option (#966)
camunda/camunda-docs-manual: content/user-guide/spring-boot-integration/develop-and-testing.md

---

title: "Developing and Testing Spring Boot Applications"
weight: 70

menu:
  main:
    name: "Developing and Testing"
    identifier: "user-guide-spring-boot-develop-and-test"
    parent: "user-guide-spring-boot-integration"

---

# Developing

> Spring provide [Developer Tools](https://docs.spring.io/spring-boot/docs/current/reference/html/using.html#using.devtools) which features options like automatic restart and live reload during the development of an application.

Reviewer: ```suggestion
Spring Boot provides [Developer Tools](https://docs.spring.io/spring-boot/docs/current/reference/html/using.html#using.devtools) which feature options like automatic restart and live reload during the development of an application.
```

## Spring Developer Tools and Classloading

> Additional process engine plugin (`ApplicationContextClassloaderSwitchPlugin`) will be loaded in case your application is in a development mode:

Reviewer: ```suggestion
An additional process engine plugin (`ApplicationContextClassloaderSwitchPlugin`) will be loaded in case your application is in a development mode:
```

* Spring Developer tools (`spring-boot-devtools` library) are on the class path and

> * the tools are enabled, e.g. the application is started in IDE

Reviewer: ```suggestion
* the tools are enabled, e.g., the application is started in IDE
```

> The plugin ensures exchanging the application context classloader with the classloader used for the process engine to prevent issues during deserialization. And the reload of resources by the Developer Tools 

Reviewer: ```suggestion
The plugin ensures exchanging the application context classloader with the classloader used for the process engine to prevent issues during deserialization.
```
Is this second sentence leftover?
